polars/polars-core/src/series/series_trait.rs

```diff
@@ -149,39 +139,37 @@ pub(crate) mod private {
             _right_column: &Series,
             _opt_join_tuples: &[(Option<IdxSize>, Option<IdxSize>)],
         ) -> Series {
-            invalid_operation_panic!(self)
+            invalid_operation_panic!(zip_outer_join_column, self)
         }
 
         fn subtract(&self, _rhs: &Series) -> PolarsResult<Series> {
-            invalid_operation_panic!(self)
+            invalid_operation_panic!(sub, self)
         }
         fn add_to(&self, _rhs: &Series) -> PolarsResult<Series> {
-            invalid_operation!(self)
+            invalid_operation_panic!(add, self)
```

Comment:
Btw, why all the panics here when most of these methods return a Result?

Reply:
Legacy reasons. If they return `Result` we should make use of that.

Reply:
Yea, should be a separate thing – to eventually get rid of as many panics (and unwraps) as possible, the first thing being panics in fallible functions. There's not many explicit panics though (compared to the number of error checks), maybe just a few hundred :) 

// There's also an enormous amount of code like e.g. "check if not none; then unwrap", or "check some condition; then unwrap" – in many of those cases unwrap is not needed if code is phrased a little differently.

Reply:
> // There's also an enormous amount of code like e.g. "check if not none; then unwrap", or "check some condition; then unwrap" – in many of those cases unwrap is not needed if code is phrased a little differently.

Yeap, the `if let Some() =` is great there. Some `unwrap()`s should stay IMO, there are good assertion and blow up in our face, so we can find the bugs.